internal/adapters/dataproviders/database/mongo/liquidity_provider.go

```diff
@@ -99,6 +105,92 @@ func (repo *lpMongoRepository) UpsertCredentials(ctx context.Context, credential
 	return upsertConfiguration(dbCtx, repo, configToStore, false)
 }
 
+func (repo *lpMongoRepository) GetTrustedAccount(ctx context.Context, address string) (*liquidity_provider.TrustedAccountDetails, error) {
+	dbCtx, cancel := context.WithTimeout(ctx, repo.conn.timeout)
+	defer cancel()
+	account := &liquidity_provider.TrustedAccountDetails{}
+	collection := repo.conn.Collection(TrustedAccountCollection)
+	filter := bson.M{"address": address}
+	err := collection.FindOne(dbCtx, filter).Decode(account)
+	if errors.Is(err, mongo.ErrNoDocuments) {
+		return nil, liquidity_provider.ErrTrustedAccountNotFound
+	} else if err != nil {
+		return nil, err
+	}
+	logDbInteraction(Read, account)
+	return account, nil
+}
+
+func (repo *lpMongoRepository) GetAllTrustedAccounts(ctx context.Context) ([]liquidity_provider.TrustedAccountDetails, error) {
+	dbCtx, cancel := context.WithTimeout(ctx, repo.conn.timeout)
+	defer cancel()
+	collection := repo.conn.Collection(TrustedAccountCollection)
+	cursor, err := collection.Find(dbCtx, bson.M{})
+	if err != nil {
+		return nil, err
+	}
+	defer cursor.Close(dbCtx)
+	var accounts []liquidity_provider.TrustedAccountDetails
+	if err = cursor.All(dbCtx, &accounts); err != nil {
+		return nil, err
+	}
+	logDbInteraction(Read, accounts)
+	return accounts, nil
+}
+
+func (repo *lpMongoRepository) AddTrustedAccount(ctx context.Context, account liquidity_provider.TrustedAccountDetails) error {
+	dbCtx, cancel := context.WithTimeout(ctx, repo.conn.timeout)
+	defer cancel()
+	collection := repo.conn.Collection(TrustedAccountCollection)
+	existingAccount, err := repo.GetTrustedAccount(ctx, account.Address)
+	if err == nil && existingAccount != nil {
```

Comment:
Besides this, you should also create a unique index, like the one we have for the pegout deposits